stdlib/3/ssl.pyi

```diff
@@ -168,6 +175,10 @@ class SSLSocket(socket.socket):
     context = ...  # type: SSLContext
     server_side = ...  # type: bool
     server_hostname = ...  # type: Optional[str]
+    if sys.version_info >= (3, 6):
+        session = ...  # type: SSLSession
```

Comment:
Should these be Optional? It looks like they're None if `self._sslobj` is None. But maybe we shouldn't worry too much about that case.

Reply:
I guess they are technically optional since they only seem to be valid for in certain states (handshaken sockets).
That distinction is not represented in the type though, so it's somewhat of a philosophical question - for objects with different states, do we take the optimistic stance (assume that the object is in the right state) or make the client code always validate the state in strict mode (`is not None`)...

Reply:
That makes sense; I don't think I have a general answer here.